app/Models/Auth.php

```diff
@@ -115,14 +115,15 @@ public static function giveAccess() {
 	 * @return boolean true if user has corresponding access, false else.
 	 */
 	public static function hasAccess($scope = 'general') {
-		$conf = Minz_Configuration::get('system');
-		$default_user = $conf->default_user;
+		$systemConfiguration = Minz_Configuration::get('system');
+		$userConfiguration = Minz_Configuration::get('user');
+		$default_user = $systemConfiguration->default_user;
 		$ok = self::$login_ok;
 		switch ($scope) {
 		case 'general':
 			break;
 		case 'admin':
-			$ok &= Minz_Session::param('currentUser') === $default_user;
+			$ok &= $default_user === Minz_Session::param('currentUser') || $userConfiguration->is_admin;
```

Comment:
It would be nice to get rid of the concept of default user here, but we would need to set the `is_admin` attribute to true for default user in some kind of migration. Several options:

- you find a quick solution and you implement it
- you wait for me to implement a migration system (I'll start today :) )
- we merge as it is, we'll perform a migration later

I have no real preference for any of these options

Reply:
We also need to make sure that a non-admin user cannot self-promote to admin, for instance during an export/import